File: server/src/main/java/com/logiflow/server/services/customer/CustomerService.java
package com.logiflow.server.services.customer;

import com.logiflow.server.dtos.customer.CustomerDtos.*;
import com.logiflow.server.models.User;

import java.util.List;

public interface CustomerService {
    User getCurrentCustomer(String authName);

    // Order management
    OrderDto createOrder(String customerUsername, CreateOrderRequest request);
    OrderDto getOrderById(String customerUsername, Integer orderId);
    List<OrderSummaryDto> getMyOrders(String customerUsername);
    TrackOrderResponse trackOrder(String customerUsername, Integer orderId);

    // Profile management
    CustomerProfileDto getProfile(String customerUsername);
    CustomerProfileDto updateProfile(String customerUsername, UpdateProfileRequest request);

    // History
    List<OrderHistoryDto> getOrderHistory(String customerUsername);
}

File: client/src/services/customerService.js
import api from './api';

const getMyOrders = async () => {
  const response = await api.get('/customer/me/orders');
  return response.data;
};

const trackOrder = async (orderId) => {
  const response = await api.get(`/customer/me/orders/${orderId}/track`);
  return response.data;
};

const customerService = {
  getMyOrders,
  trackOrder,
};

export default customerService;
